app/src/main/java/org/mozilla/focus/components/StoreProvider.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.components
+
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.get
+import mozilla.components.lib.state.Store
+
+/**
+ * Generic ViewModel to wrap a State object for state restoration.
+ *
+ * @property store [Store] instance attached to [ViewModel].
+ */
+class StoreProvider<T : Store<*, *>>(
+    val store: T
+) : ViewModel() {
+
+    companion object {
+        fun <T : Store<*, *>> get(fragment: Fragment, createStore: () -> T): T {
+            val factory = StoreProviderFactory(createStore)
+            val viewModel: StoreProvider<T> = ViewModelProvider(fragment, factory).get()
+            return viewModel.store
+        }
+    }
+}
+
+/**
+ * ViewModel factory to create [StoreProvider] instances.
+ *
+ * @param createStore Callback to create a new [Store], used when the [ViewModel] is first created.
+ */
+class StoreProviderFactory<T : Store<*, *>>(
+    private val createStore: () -> T
+) : ViewModelProvider.Factory {
+
+    @Suppress("UNCHECKED_CAST")
+    override fun <VM : ViewModel?> create(modelClass: Class<VM>): VM {
+        return StoreProvider(createStore()) as VM
+    }
+}
```

Comment:
Attaching the `Store` to a `ViewModel` is an interesting idea. But I wonder if it is needed. If we want to keep track of a `Store` instance across recompositions then we could/should use `remember()`.

To create a `Store` that saves/restores it state through configuration changes, and including process restores, we have a `composableStore()` helper [in A-C](https://github.com/mozilla-mobile/android-components/blob/main/components/lib/state/src/main/java/mozilla/components/lib/state/ext/ComposeExtensions.kt#L84).

Since all data the locale store holds is already persisted, it seems like we could away with just using `remember()` for recompositions and otherwise, on configuration changes, can live with the store (and its state) getting recreated?

Reply:
org.mozilla.fenix.components.StoreProvider is the same class like in Fenix  . It is used in all the fragment there . I think is a good idea to move the class in A-C , to eliminate  duplicated code.

Reply:
I think that emerged early in Fenix when we wanted to keep the state of a `Store` through configuration changes. For a `View`-based app using a `ViewModel` seems like the most convenient way. Now, with Jetpack Compose, we may have other patterns/mechanisms that are not as heavy as a `ViewModel`. Here the `ViewModel` only acts as a container that holds the `Store`. There are more low-level APIs of Compose that we could use (some examples above). But I think that we do not need to solve this now - something similar will have to happen in Fenix at some point too. (CC @csadilek, just to be aware of this).

In this particular case I wonder if we could not do anything fancy at all and just create a Store and attach it to `LanguageFragment`. It seems like there's no state we need to keep around since it can be recreated from the storage anyways?